samples/net/openthread/coprocessor/src/main.c

```diff
@@ -12,39 +12,50 @@ LOG_MODULE_REGISTER(ot_br, LOG_LEVEL_DBG);
 #include <zephyr.h>
 #include <usb/usb_device.h>
 #include <drivers/uart.h>
+#include <net/openthread.h>
+#include <openthread/ncp.h>
 
 #define APP_BANNER "***** OpenThread NCP on Zephyr %s *****"
 
 void main(void)
 {
-#if defined(CONFIG_OPENTHREAD_COPROCESSOR_SPINEL_ON_UART_ACM)
-	const struct device *dev;
-	uint32_t baudrate = 0U;
-	int ret;
-
-	dev = device_get_binding(
-		CONFIG_OPENTHREAD_COPROCESSOR_SPINEL_ON_UART_DEV_NAME);
-	if (!dev) {
-		LOG_ERR("UART device not found");
-		return;
+	if (IS_ENABLED(CONFIG_OPENTHREAD_COPROCESSOR_SPINEL_ON_UART_ACM)) {
+		int ret;
+		const struct device *uart_dev;
+		uint32_t dtr = 0U;
+
+		ret = usb_enable(NULL);
+		if (ret != 0) {
+			LOG_ERR("Failed to enable USB");
+			return;
+		}
+
+		uart_dev = device_get_binding(
+			CONFIG_OPENTHREAD_COPROCESSOR_SPINEL_ON_UART_DEV_NAME);
+
+		if (uart_dev == NULL) {
+			LOG_ERR("UART device not found");
+			return;
+		}
+
+		/* Data Terminal Ready - check if host is ready to communicate */
+		while (!dtr) {
+			ret = uart_line_ctrl_get(uart_dev, UART_LINE_CTRL_DTR,
+						 &dtr);
+			if (ret) {
+				LOG_ERR("Failed to get Data Terminal Ready line state: %d",
+					ret);
+				continue;
+			}
```

Comment:
Would it make sense to add a small delay (k_sleep) between the iterations? We're practically busy-looping right now.

Reply:
I don't think it will change much. Currently any more important things happen with higher priorities and because main is working as preemptive it can be supplanted. Anyways the application is useless until it connects with a host for what we waits here.

Reply:
On the other hand small delay here would let logging thread to process... Ok, let's do that.